addons/stock_account/models/product.py

```diff
@@ -440,11 +450,12 @@ def _run_fifo_vacuum(self, company=None):
                 'stock_valuation_layer_id': svl_to_vacuum.id,
             }
             vacuum_svl = self.env['stock.valuation.layer'].sudo().create(vals)
+            # recompute it manually to avoid using heavy _compute_value_svl
+            value_svl += corrected_value
 
             # If some negative stock were fixed, we need to recompute the standard price.
-            product = self.with_company(company.id)
-            if product.cost_method == 'average' and not float_is_zero(product.quantity_svl, precision_rounding=self.uom_id.rounding):
-                product.sudo().with_context(disable_auto_svl=True).write({'standard_price': product.value_svl / product.quantity_svl})
+            if product.cost_method == 'average' and not float_is_zero(quantity_svl, precision_rounding=self.uom_id.rounding):
```

Comment:
You can't just put it outside the loop and recompute it only at the end instead. I don't think we need it inside the loop.